File: src/main/kotlin/com/github/sydowma/fastplugin/settings/SettingsComponent.java
package com.github.sydowma.fastplugin.settings;

import javax.swing.*;
import java.awt.*;

public class SettingsComponent {
    private JPanel mainPanel;
    private JTextField apiKeyField;
    private JTextArea promptField;
    private JComboBox<String> defaultPromptComboBox;

    public SettingsComponent() {
        mainPanel = new JPanel(new BorderLayout());
        apiKeyField = new JTextField();
        promptField = new JTextArea();
        defaultPromptComboBox = new JComboBox<>();

        mainPanel.add(new JLabel("OpenAI API Key:"), BorderLayout.NORTH);
        mainPanel.add(apiKeyField, BorderLayout.NORTH);
        mainPanel.add(new JLabel("Prompts:"), BorderLayout.CENTER);
        mainPanel.add(new JScrollPane(promptField), BorderLayout.CENTER);
        mainPanel.add(new JLabel("Default Prompt:"), BorderLayout.SOUTH);
        mainPanel.add(defaultPromptComboBox, BorderLayout.SOUTH);
    }

    public JPanel getPanel() {
        return mainPanel;
    }

    public boolean isModified() {
        // Implement logic to check if settings are modified
        return true;
    }

    public void apply() {
        // Implement logic to save settings
    }

    public void reset() {
        // Implement logic to reset settings
    }
}

File: src/main/kotlin/com/github/sydowma/fastplugin/settings/SettingsComponent.kt
package com.github.sydowma.fastplugin.settings

import java.awt.BorderLayout
import javax.swing.*

class SettingsComponent {
    val panel: JPanel = JPanel(BorderLayout())
    private val apiKeyField = JTextField()
    private val promptField = JTextArea()
    private val defaultPromptComboBox = JComboBox<String>()

    init {
        panel.add(JLabel("OpenAI API Key:"), BorderLayout.NORTH)
        panel.add(apiKeyField, BorderLayout.NORTH)
        panel.add(JLabel("Prompts:"), BorderLayout.CENTER)
        panel.add(JScrollPane(promptField), BorderLayout.CENTER)
        panel.add(JLabel("Default Prompt:"), BorderLayout.SOUTH)
        panel.add(defaultPromptComboBox, BorderLayout.SOUTH)
    }

    fun isModified(settings: SettingsState): Boolean {
        return (apiKeyField.text != settings.apiKey
                || promptField.text != settings.prompt
                || defaultPromptComboBox.selectedItem != settings.defaultPrompt)
    }

    fun apply(settings: SettingsState) {
        settings.apiKey = apiKeyField.text
        settings.prompt = promptField.text
        settings.defaultPrompt = (defaultPromptComboBox.selectedItem as String)
    }

    fun reset(settings: SettingsState) {
        apiKeyField.text = settings.apiKey
        promptField.text = settings.prompt
        defaultPromptComboBox.selectedItem = settings.defaultPrompt
    }
}

